src/Orleans/Runtime/GrainRuntime.cs

```diff
@@ -8,18 +8,15 @@ namespace Orleans.Runtime
     internal class GrainRuntime : IGrainRuntime
     {
         private readonly string id;
-        private readonly IGrainFactory grainFactory;
-        private readonly ITimerRegistry timerRegistry;
-        private readonly IReminderRegistry reminderRegistry;
-        private readonly IStreamProviderManager streamProviderManager;
 
-        public GrainRuntime(string id, IGrainFactory grainFactory, ITimerRegistry timerRegistry, IReminderRegistry reminderRegistry, IStreamProviderManager streamProviderManager)
+        public GrainRuntime(string id, IGrainFactory grainFactory, ITimerRegistry timerRegistry, IReminderRegistry reminderRegistry, IStreamProviderManager streamProviderManager, IStorage storage=null)
```

Comment:
Why storage can be null? This is relying on a logic of assigning a default storage in the `Grain`. This hurts code readability as it's impossible for someone new looking at this class to know what's going on.